### ISSUE TITLE
error in enthalpy of PR

### ISSUE THREAD

**Issue by KSTU** [ClapeyronThermo/Clapeyron.jl#317]
I calculate the enthalpy of methane and compare it with the data from the DWsim and Hysys programs. The results for the enthalpy values ​​do not match.
```
	model = PR(["methane"])
	H_mol = enthalpy(model, 101325, 298.15, [1.0])	#J/mol
	m = model.params.Mw[1] / 1000
	H_mass = enthalpy(model, 101325, 298.15, [1.0]) / m	#J/kg
	println(H_mass )
```
give 385222.4290570486 
DWsim give -1146.13 J/kg
![Screenshot from 2024-12-04 21-38-43](https://github.com/user-attachments/assets/ad70fb94-f186-4409-afd2-2680aeb775f8)

**Comment by longemen3000** [comment]
Hello,

This problem actually encompasses two different concepts and a bug:
1. Our default model is known as `BasicIdeal`. it supposes Cp = 2.5R. this suffices for light gases and it is enough for models that don't use caloric properties, but you need to specify an adequate caloric model. in this particular case, i recommend `ReidIdeal`, but if you are working with only natural gas components, you could also use `AlyLeeIdeal` (the same Cp that GERG2008 uses in the ideal gas limit)
2. enthalpies, entropies, internal energies and other energy expressions are defined in terms of a reference state. this reference state is arbitrary. so you need to select the correct reference state to compare. for phase equilibria and caloric applications, the reference state does not matter, the only important thing is the differences between enthalpies. I couldnt find what reference state does DWSim uses, HYSYS seems to use the ideal gas heat of formation at standard temperature and pressure. you can check the what the CoolProp docs say about reference states http://www.coolprop.org/coolprop/HighLevelAPI.html#high-level-set-reference-state
3. This is a bug in our current release that i found while testing your problem, but, at this moment, the following code does not work:
```julia
model = PR("ethane",idealmodel = ReidIdeal, reference_state = :ashrae)
```
i'm fixing this and releasing a new version as soon as possible.

**Comment by KSTU** [comment]
Thank you.
The difference was discovered when calculating the heat exchanger. Precisely because the difference in enthalpy gave incorrect outlet temperatures. There are also deviations in entropy, although it should not depend on the selected referencs energy level.
For now we will use only on Cool Prop, waiting for a new release.

**Comment by longemen3000** [comment]
Try using `PR("ethane", idealmodel = ReidIdeal)`, that should fix the enthalpy and entropy differences. Entropy values also have a dependence on the reference state, but entropy differences do not.

**Comment by Sush1090** [comment]
Just a small addition to this discussion:

Some time ago I was trying to find which Clapyeron.jl model is closest to CoolProp. 

The conclusion I have for now is `cPR` with `ReidIdeal` has show the closest results for most single component systems. The best variable to compare is `isobaric_heat_capacity`

```julia

using Clapeyron
using CoolProp
function mass_to_moles(model::EoSModel,compositions::AbstractVector,mass)
    @assert isapprox(sum(compositions),1.0,atol = 1e-8)
    mws = Clapeyron.mw(model);
    M = mass*compositions;
    return M./mws
end

model_reid = cPR(["methane"],idealmodel = ReidIdeal)
model_basic = cPR(["methane"],idealmodel = BasicIdeal)
mass = 1000 #grams
z = mass_to_moles(model_reid,[1],mass)
Cp_clapeyron_reid = isobaric_heat_capacity(model_reid, 101325, 298.15, z)
Cp_clapeyron_basic = isobaric_heat_capacity(model_basic, 101325, 298.15, z)


Cp_coolprop = PropsSI("CPMASS","P",101325,"T",298.15,"methane")

@show rel_error_reid = abs(Cp_clapeyron_reid - Cp_coolprop)/abs(Cp_clapeyron_reid)
@show rel_error_basic = abs(Cp_clapeyron_basic - Cp_coolprop)/abs(Cp_clapeyron_basic)
```

```julia
rel_error_reid = abs(Cp_clapeyron_reid - Cp_coolprop) / abs(Cp_clapeyron_reid) = 0.0020650075616607437
rel_error_basic = abs(Cp_clapeyron_basic - Cp_coolprop) / abs(Cp_clapeyron_basic) = 0.7142500577988615
```

**Comment by longemen3000** [comment]
solved in 0.6.6
